src/pkg/services/m365/api/events.go

```diff
@@ -153,24 +154,31 @@ func (c Events) GetContainerByName(
 		return nil, graph.Stack(ctx, err).WithClues(ctx)
 	}
 
-	// We only allow the api to match one calendar with provided name.
-	// Return an error if multiple calendars exist (unlikely) or if no calendar
-	// is found.
-	if len(resp.GetValue()) != 1 {
-		err = clues.New("unexpected number of calendars returned").
-			With("returned_calendar_count", len(resp.GetValue()))
-		return nil, err
+	gv := resp.GetValue()
+
+	if len(gv) == 0 {
+		return nil, clues.New("container not found").WithClues(ctx)
+	}
+
+	// Return an error if no calendar is found.
+	if len(resp.GetValue()) == 0 {
+		return nil, clues.New("unexpected number of calendars returned").
+			With("returned_container_count", len(gv)).
+			WithClues(ctx)
 	}
 
+	// We only allow the api to match one calendar with the provided name.
+	// If we match multiples, we'll eagerly return the first one.
```

Comment:
just for the sake of debuggability could you add a log call with the number of results returned. It can be debug level